test/grouping.jl

```diff
@@ -3951,4 +3951,227 @@ end
+    @test combine(gdf2, groupindices, groupindices => :gidx,
+                  proprow, proprow => :freq) ==
+          DataFrame(id=["b", "a"],
+                    groupindices=[1, 2],
+                    gidx=[1, 2],
+                    proprow=[2/5, 3/5],
+                    freq=[2/5, 3/5])
+    gdf3 = gdf2[[1]]
+    @test combine(gdf3, eachindex, eachindex => :idx,
+                  groupindices, groupindices => :gidx,
+                  proprow, proprow => :freq) ==
+          DataFrame(id=["b", "b"],
+                    eachindex=[1, 2],
+                    idx=[1, 2],
+                    groupindices=[1, 1],
+                    gidx=[1, 1],
+                    proprow=[1.0, 1.0],
+                    freq=[1.0, 1.0])
+    @test combine(gdf3, groupindices, groupindices => :gidx,
+                  proprow, proprow => :freq) ==
+          DataFrame(id=["b"],
+                    groupindices=[1],
+                    gidx=[1],
+                    proprow=[1.0],
+                    freq=[1.0])
+    gdf4 = gdf3[[false]]
+    @test isequal_coltyped(combine(gdf4, eachindex, eachindex => :idx,
+                                   groupindices, groupindices => :gidx,
+                                   proprow, proprow => :freq),
+                           DataFrame(id=String[], eachindex=Int[], idx=Int[],
+                                     groupindices=Int[], gidx=Int[],
+                                     proprow=Float64[], freq=Float64[]))
+    @test isequal_coltyped(combine(gdf4, groupindices, groupindices => :gidx,
+                                   proprow, proprow => :freq),
+                           DataFrame(id=String[], groupindices=Int[], gidx=Int[],
+                                     proprow=Float64[], freq=Float64[]))
+    gdf5 = groupby(DataFrame(), [])
+    @test isequal_coltyped(combine(gdf5, eachindex, eachindex => :idx,
+                                   groupindices, groupindices => :gidx,
+                                   proprow, proprow => :freq),
+                           DataFrame(eachindex=Int[], idx=Int[],
+                                     groupindices=Int[], gidx=Int[],
+                                     proprow=Float64[], freq=Float64[]))
+    @test isequal_coltyped(combine(gdf5, groupindices, groupindices => :gidx,
+                                   proprow, proprow => :freq),
+                           DataFrame(groupindices=Int[], gidx=Int[],
+                                     proprow=Float64[], freq=Float64[]))
+
+    # eachindex on DataFrame
+    @test combine(df, eachindex) == DataFrame(eachindex=1:6)
+    @test isequal_coltyped(combine(DataFrame(), eachindex),
+                           DataFrame(eachindex=Int[]))
+
+    # Disallowed operations
+    @test_throws ArgumentError groupindices(df)
+    @test_throws ArgumentError proprow(df)
+    @test_throws ArgumentError combine(df, groupindices)
+    @test_throws ArgumentError combine(df, proprow)
```

Comment:
I guess one could make the argument that these should work and treat the data frame as a single group, like we do elsewhere. But that can be changed later if needed.

Reply:
Yes - I was considering this. However, I came to the conclusion that it is better to strongly signal in the documentation that `groupindices` and `proprow` are `GroupedDataFrame` specific.

In particular remember that:

> treat the data frame as a single group

is not correct for 0-row data frames, which are treated as having 0 groups.

For this reason I have decided, that for now, it is better not to support it as both these functions operations are not really useful for `DataFrame` as a source I think (but I can add it if you disagree).